### PR TITLE
Add gpg_fingerprint lookup and filter

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
The lookup returns the (long) fingerprint of a GPG key from a public or private key file.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
gpg_fingerprint

### CODE REVIEW

File: tests/integration/targets/lookup_gpg_fingerprint/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      ignore_errors: true
+
+    - name: Check result
+      ansible.builtin.assert:
+        that:
+          - result is failed
+          - >-
+            (remote_tmp_dir ~ '/does-not-exist does not exist') in result.msg
+
+    - name: Write garbage to disk
```
Comment (briantist) [inline tests/integration/targets/lookup_gpg_fingerprint/tasks/main.yml:75]: me, every time I hit save in my IDE 😏

**Review by briantist — APPROVED** [review]

File: plugins/lookup/gpg_fingerprint.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+name: gpg_fingerprint
+short_description: Retrieve a GPG fingerprint from a GPG public or private key file
+author: Felix Fontein (@felixfontein)
+version_added: 2.15.0
+description:
+  - "Takes a list of filenames pointing to GPG public or private key files. Returns the fingerprint for every of these keys."
```
Comment (samccann) [inline plugins/lookup/gpg_fingerprint.py:15]:
```suggestion
  - "Takes a list of filenames pointing to GPG public or private key files. Returns the fingerprints for each of these keys."
```

**Review by samccann — APPROVED** [review]
docs portion lgtm

**Comment by felixfontein** [comment]
If nobody objects to the way the lookup and filter work and how they are implemented, I'll merge this in ~a week.

**Comment by felixfontein** [comment]
@briantist @samccann thanks a lot for reviewing!

**Pull request merged**